test/OrleansTestKit.Tests/Tests/StorageFacetTests.cs

```diff
@@ -19,15 +20,7 @@ public class StorageFacetTests : TestKitBase
         public async Task GetColor_WithDefaultState_ReturnsUnknown()
         {
             // Arrange
-            var state = new ColorGrainState();
-
-            var mockState = new Mock<IPersistentState<ColorGrainState>>();
-            mockState.SetupGet(o => o.State).Returns(state);
-
-            var mockMapper = new Mock<IAttributeToFactoryMapper<PersistentStateAttribute>>();
-            mockMapper.Setup(o => o.GetFactory(It.IsAny<ParameterInfo>(), It.IsAny<PersistentStateAttribute>())).Returns(context => mockState.Object);
-
-            Silo.AddService(mockMapper.Object);
+            Silo.AddPersistentState<ColorGrainState>();
```

Comment:
This PR opens up two new possibilities, and I think we should document it with new test cases or a brand new test suite:

- A grain that depends on multiple states using the facets/DI.
- A grain that extends `Grain<TState>` _and_ has a second state using the facets/DI.

Reply:
I added some more tests for a grain using both `Grain<TState` and the facets. I didn't have time to add tests with multiple states although that should work without a problem.